### ISSUE TITLE
Enum as a query variable causes validation errors

### ISSUE THREAD

**Issue by dotellie** [nestjs/graphql#1001]
## I'm submitting a...

<pre><code>
[x] Regression 
[ ] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

Using an input with an enum type on a query as a variable causes a validation error. A query like this does not work as an example if `UserType` is an enum:

```gql
mutation ($userType: UserType!) {
  addUser(type: $userType) {
    id
    type
  }
}
```

The query would yield the following:

```
"GraphQLError: Variable \"$userType\" of type \"UserType!\" used in position expecting type \"UserType!\"." [...]
```

## Expected behavior

The above query should work without any problems.

## Minimal reproduction of the problem with instructions

Reused my old repro repo again, the query above is from its current state: https://github.com/dotellie/nest-gql-resolve-field-repro

## Regression
The issue started appearing on 7.4.4 and after so it's likely a problem with the custom schema transforms if I had to guess.

**Comment by dotellie** [comment]
After spending a bit time on trying to fix this, I realized it's also an issue for all places where arguments can be provided and it's probably also an issue with input types as well as enums. Basically all types which can be used as inputs are duplicated in the schema. I'm likely going to be able to fix this and do a PR, but I'm probably going the route of copying all of Apollo's `transformSchema` and patching it to not remove additional fields on the objects.

**Comment by kamilmysliwiec** [comment]
Fix published as 7.5.1. :)
